quickwit-indexing/src/actors/indexer.rs

```diff
@@ -397,12 +404,26 @@ impl Indexer {
         &mut self,
         commit_trigger: CommitTrigger,
         ctx: &ActorContext<Self>,
-    ) -> Result<(), SendError> {
+    ) -> anyhow::Result<()> {
         let indexed_split = if let Some(indexed_split) = self.current_split_opt.take() {
             indexed_split
         } else {
             return Ok(());
         };
+
+        // Avoid sending empty split.
```

Comment:
```suggestion
        // Avoid producing empty split, but still update the checkpoint to avoid
        // reprocessing the same faulty documents.
```